Test/AblyTests/Tests/MessageUpdatesDeletesTests.swift

```diff
@@ -0,0 +1,606 @@
+                    done()
+                }
+            }
+        }
+        
+        XCTAssertNotNil(publishedMessage)
+        
+        // Update the message to create versions
+        waitUntil(timeout: testTimeout) { done in
+            channel.update(publishedMessage!.serial!, data: "updated data") { error in
+                XCTAssertNil(error)
+                done()
+            }
+        }
+        
+        // Get the message versions
+        waitUntil(timeout: testTimeout) { done in
+            channel.getMessageVersions(publishedMessage!.serial!) { result, error in
+                versionsResult = result
+                versionsError = error
+                done()
+            }
+        }
+        
+        XCTAssertNil(versionsError)
+        XCTAssertNotNil(versionsResult)
+        // Should have at least 2 versions: original and updated
+        XCTAssertGreaterThanOrEqual(versionsResult?.items.count ?? 0, 2)
+    }
+    
+    // MARK: - Encoding/Decoding Tests
+    
+    // Test that delete is implemented as an update
+    func test__016__REST__delete__should_be_implemented_as_update() throws {
+        let test = Test()
+        let testEnvironment = try TestEnvironment(test: test)
+        
+        var publishedMessage: ARTMessage?
+        var deleteError: ARTErrorInfo?
+        
+        let channel = testEnvironment.client.channels.get(test.uniqueChannelName())
+        
+        // First publish a message
+        waitUntil(timeout: testTimeout) { done in
+            channel.publish("test-event", data: "data to delete") { error in
+                XCTAssertNil(error)
+                
+                channel.history { result, _ in
+                    publishedMessage = result?.items.first
+                    done()
+                }
+            }
+        }
+        
+        XCTAssertNotNil(publishedMessage)
+        
+        // Delete the message (which is implemented as an update with action=Delete)
+        waitUntil(timeout: testTimeout) { done in
+            channel.delete(publishedMessage!.serial!) { error in
+                deleteError = error
+                done()
+            }
+        }
+        
+        XCTAssertNil(deleteError)
+        
+        // Retrieve the message to verify it has delete action
+        var retrievedMessage: ARTMessage?
+        waitUntil(timeout: testTimeout) { done in
+            channel.get(publishedMessage!.serial!) { message, error in
+                retrievedMessage = message
+                done()
+            }
+        }
+        
+        // The message should still exist but with action=Delete
+        XCTAssertNotNil(retrievedMessage)
+        XCTAssertEqual(retrievedMessage?.action, .delete)
+    }
+    
+    // Test encoding message with action, serial, version, and annotations
+    func test__017__Encoding__message__should_encode_all_properties() throws {
+        let message = ARTMessage(name: "test", data: "test data")
```

Comment:
What's going on with this test file? It seems to be a mixture of integration tests (with no spec point references) and unit tests for things that aren't related to the new functionality.

Also why are the tests numbered? The only reason we did that in our existing tests was because when we got rid of Quick we needed to preserve the order of test execution because some of those tests depended on the execution order. We shouldn't be introducing new tests that depend on execution order.

Reply:
Replaced with normal one (contains only one test yet).